### PR TITLE
[Bugfix]support query external_catalog table when current_catalog is default_catalog

### PR DESCRIPTION
**Pull request by stephen-shelby**
support query external_catalog table when current_catalog is default_catalog

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [x] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

- `TableName#tosql()` add `catalog` field
- `AstBuilder#visitColumnReference()` adapt to column names which has four level.

### CODE REVIEW

**Comment by stephen-shelby** [comment]
cc @HangyuanLiu

**Review by HangyuanLiu — APPROVED** [review]

**Comment by stephen-shelby** [comment]
@Mergifyio backport branch-2.3

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ private void getDB(TableName tableName) {
             if (Strings.isNullOrEmpty(dbName)) {
                 dbName = session.getDatabase();
             } else {
-                if (CatalogMgr.isInternalCatalog(session.getCurrentCatalog())) {
+                if (CatalogMgr.isInternalCatalog(tableName.getCatalog())) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java:204]: MetaUtils.normalizationTableName?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|11|11|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java:204]: it has executed normalTableName here.  it cover the case that query connector table when current_catalog is default_catalog.

**Review by caneGuy — APPROVED** [review]

**Pull request merged**